src/cli/one_helper/onezone_helper.rb

```diff
@@ -14,8 +14,373 @@
+
+        # backup oned.conf
+        FileUtils.cp('/etc/one/oned.conf',
+                        "/etc/one/oned.conf#{time_based_identifier}")
+
+        FED_ATTRS.zip(fed_attrs) do |name, value|
+            r_aug.set("FEDERATION/#{name}", value)
+        end
+
+        r_aug.save
+
+        FileUtils.cp(r_oned.path, '/etc/one/oned.conf')
+
+        puts 'oned.conf has been replaced by ' \
+                "#{@remote_server}:/etc/one/oned.conf"
+
+        puts 'A copy of your old oned.conf file is located here: ' \
+                "/etc/one/oned.conf#{time_based_identifier}"
+
+        @opennebula_services['opennebula'] = true
+    end
+
+    # Sync database
+    def sync_db
+        puts "Dumping and fetching database from #{@remote_server}, " \
+                'this could take a while'
+
+        SSHCommand.run(
+            "onedb backup -f -u #{@r_config_elements[:user]} " \
+            "-p #{@r_config_elements[:password]} " \
+            "-d #{@r_config_elements[:dbname]} " \
+            "-P #{@r_config_elements[:port]} /tmp/one_db_dump.sql",
+            @remote_server
+        )
+
+        run_command(
+            "scp -i #{@oneadmin_identity_file} #{SSH_OPTIONS} " \
+            "#{@remote_server}:/tmp/one_db_dump.sql /tmp/one_db_dump.sql"
+        )
+
+        puts "Local OpenNebula's database will be replaced, hang tight"
+
+        service_action('opennebula', 'stop')
+
+        puts 'Restoring database'
+
+        run_command(
+            "onedb restore -f -u #{@l_config_elements[:user]} " \
+                "-p #{@l_config_elements[:password]} " \
+                "-d #{@l_config_elements[:dbname]} " \
+                "-P #{@l_config_elements[:port]} " \
+                "-h #{@l_config_elements[:server]}" \
+                '/tmp/one_db_dump.sql'
+        )
+
+        service_action('opennebula', 'start')
+    end
+
+    # Run local command
+    #
+    # @param cmd [String] Command to run
+    def run_command(cmd)
+        output = LocalCommand.run(cmd)
+
+        if output.code == 0
+            output
+        else
+            STDERR.puts 'ERROR'
+            STDERR.puts "Failed to run: #{cmd}"
+            STDERR.puts output.stderr
+
+            exit(-1)
+        end
+    end
+
+    # Restart OpenNebula services
+    def restart_services
+        restarted = false
+
+        @opennebula_services.each do |service, status|
+            next unless status
+
+            service_action(service)
+
+            restarted = true
+        end
+
+        return if restarted
+
+        puts 'Everything seems synchronized, nothing was replaced.'
+    end
+
+    # Service action
+    #
+    # @param service [String] Service to restart
+    # @param action  [String] Action to execute (start, stop, restart)
+    def service_action(service, action = 'restart')
+        if `file /sbin/init`.include? 'systemd'
+            puts "Restarting #{service} via systemd"
+            run_command("systemctl #{action} #{service}")
```

Comment:
Better `restart` -> `try-restart` OR you have to check first if the service is running. Also, the case when Apache is running within Apache is not handled (at least with `try-restart` won't start another Sunstone instance).